### DISCUSSION TITLE
How can I set path to .cpt files in gauge?

### DISCUSSION THREAD

**Discussion by flexin777** [getgauge/gauge#1882] [General]
Hello
I see that there is setting for steps implementation path in js.properties.
STEP_IMPL_DIR = steps

Is there a possibility to add path to concepts(.cpt) files?

**Comment by zabil** [comment]
From Gauge version 1.1.5 and above it will look for concept files at `STEP_IMPL_DIR`. There is no separate configuration to set only the concept file paths.

**Reply by flexin777** [reply]
Thanks for your answer!
Could you tell me, please?
What the logic is how Gauge finds the concept files. What are the rules for Gauge of finding the concept-files? Where must they be and where must they not be relatively from the spec-files?

**Reply by sriv** [reply]
Gauge scans the entire project folder (including the specs directory) for files with `.cpt` extension:

The below directories are ignored in this scan:

- gauge_bin
- reports
- logs
- env
- location set in `gauge_reports_dir` env
- location set in `logs_directory` env
- locations listed in `gauge_exclude_dirs` env

**Reply by flexin777** [reply]
Thank you for an answer. 
One more question. 
Is it possible somehow to tell Gauge to look for .cpt files in folder that is one level up(from folder_child will look for .cpt files and steps in folder_parent)

folder_parent
  +folder_child

**Reply by flexin777** [reply]
I would like to have all *.spec files in one folder and steps&cpt files in another (parent) folder.

**Reply by sriv** [reply]
No, currently this is not possible. Gauge expects the concepts to be inside the project root.
